File: Dynamic Programming Applications In Machine Learning and Genomics/Alignment with Affine Gap Penalties/affine_gap_penalties.py
#python3
import sys

def align(m,mu,sigma,eps,s,t):
    """
    m, mu, and sigma are the scoring parameters (eps is epsilon)
    s and t are the strings to be aligned

    return a string with the score and reconstructed alignment
    be sure to follow the output format for this problem
    """
    # TODO: your code here
    return ""

if __name__ == "__main__":
    m,mu,sigma,eps = map(int,sys.stdin.readline().strip().split())
    s,t = [sys.stdin.readline().strip() for _ in range(2)]
    print(align(m,mu,sigma,eps,s,t))

File: Dynamic Programming Applications In Machine Learning and Genomics/Alignment with Affine Gap Penalties/affine_gap_penalties.cpp
#include <iostream>
#include <string>
#include <algorithm>

using std::cout;
using std::cin;
using std::endl;
using std::string;

/**
 * m, mu, sigma, and eps are the scoring parameters (eps is epsilon)
 * s and t are the strings to be aligned 
 *
 * return a string with the alignment score and the reconstructed alignment
 * be sure to follow the problem's output format
 */
string align(int m, int mu, int sigma, int eps, const string& s, const string& t) {
    // TODO: your code here
    return "";
}


int main() {
    int m,mu,sigma,eps;
    cin >> m >> mu >> sigma >> eps;
    cin.ignore();
    string s,t;
    getline(cin, s);
    getline(cin, t);

    cout << align(m,mu,sigma,eps,s,t) << endl;
    return 0;
}
